packages/dds/tree/src/test/edit-manager/editManger.spec.ts

```diff
@@ -0,0 +1,606 @@
+
+interface TestChangeset {
+    inputContext?: number;
+    outputContext?: number;
+    intentions: number[];
+}
+
+interface NonEmptyTestChangeset extends TestChangeset {
+    inputContext: number;
+    outputContext: number;
+}
+
+function isNonEmptyChange(change: RecursiveReadonly<TestChangeset>):
+	change is RecursiveReadonly<NonEmptyTestChangeset> {
+    return change.inputContext !== undefined && change.outputContext !== undefined;
+}
+
+class TestChangeRebaser implements ChangeRebaser<TestChangeset, TestChangeset, TestChangeset> {
+    private contextCounter: number = 0;
+    private intentionCounter: number = 0;
+
+    public compose(...changes: TestChangeset[]): TestChangeset {
+        let inputContext: number | undefined;
+        let outputContext: number | undefined;
+        const intentions: number[] = [];
+        for (const change of changes) {
+            inputContext ??= change.inputContext;
+            outputContext = change.outputContext ?? outputContext;
+            intentions.push(...change.intentions);
+        }
+        return {
+            inputContext,
+            outputContext,
+            intentions,
+        };
+    }
+
+    public invert(change: TestChangeset): TestChangeset {
+        return {
+            inputContext: change.outputContext,
+            outputContext: change.inputContext,
+            intentions: change.intentions.map((i) => -i),
+        };
+    }
+
+    public rebase(change: TestChangeset, over: TestChangeset): TestChangeset {
+        return {
+            inputContext: over.outputContext,
+            outputContext: ++this.contextCounter,
+            intentions: change.intentions,
+        };
+    }
+
+    public rebaseAnchors(anchor: AnchorSet, over: TestChangeset): void {
+        throw new Error("Method not implemented.");
+    }
+
+    public import(change: TestChangeset): TestChangeset {
+        return change;
+    }
+
+    public export(change: TestChangeset): TestChangeset {
+        return change;
+    }
+
+    public mintChangeset(inputContext: number): NonEmptyTestChangeset {
+        return {
+            inputContext,
+            outputContext: ++this.contextCounter,
+            intentions: [++this.intentionCounter],
+        };
+    }
+
+    public checkChangeList(changes: readonly RecursiveReadonly<TestChangeset>[], intentions?: Set<number>): void {
+        const filtered = changes.filter(isNonEmptyChange);
+        const intentionsSeen = new Set<number>();
+        const intentionsExpected = new Set<number>(
+            intentions ??
+            makeArray(this.intentionCounter, (i: number) => i + 1),
+        );
+        let index = 0;
+        for (const change of filtered) {
+            for (const intention of change.intentions) {
+                if (intention > 0) {
+                    // The same intention should never be applied multiple times
+                    assert.strictEqual(intentionsSeen.has(intention), false);
```

Comment:
All these boolean strict equality asserts should probably be replaced with some boolean assert call.

Reply:
I can't seem to find such a call on `assert`. 0_0

Reply:
It looks like you should be able to call `assert()` or `assert.ok()`.